lib/dal/src/workspace_snapshot/node_weight/content_node_weight.rs

```diff
@@ -177,4 +185,110 @@ impl From<DeprecatedContentNodeWeightV1> for ContentNodeWeight {
     }
 }
 
-impl CorrectTransforms for ContentNodeWeight {}
+fn remove_outgoing_prototype_argument_value_targets_except(
+    graph: &WorkspaceSnapshotGraphV3,
+    prototype_idx: NodeIndex,
+    source: NodeInformation,
+    destination_component_id: ComponentId,
+) -> impl Iterator<Item = Update> + '_ {
+    graph
+        .edges_directed(prototype_idx, Outgoing)
+        .filter(move |edge_ref| {
+            EdgeWeightKindDiscriminants::PrototypeArgument == edge_ref.weight().kind().into()
+        })
+        .filter_map(move |edge_ref| {
+            graph
+                .get_node_weight_opt(edge_ref.target())
+                .and_then(|weight| match weight {
+                    NodeWeight::AttributePrototypeArgument(apa_inner) => {
+                        apa_inner.targets().and_then(|targets| {
+                            (targets.destination_component_id == destination_component_id)
+                                .then_some(weight)
+                        })
+                    }
+                    _ => None,
+                })
+                .map(|target_weight| Update::RemoveEdge {
+                    source,
+                    destination: target_weight.into(),
+                    edge_kind: EdgeWeightKindDiscriminants::PrototypeArgument,
+                })
+        })
+}
+
+impl CorrectTransforms for ContentNodeWeight {
+    fn correct_transforms(
+        &self,
+        graph: &WorkspaceSnapshotGraphV3,
+        mut updates: Vec<Update>,
+        _from_different_change_set: bool,
+    ) -> CorrectTransformsResult<Vec<Update>> {
+        if self.content_address_discriminants() != ContentAddressDiscriminants::AttributePrototype {
+            return Ok(updates);
+        }
+
+        let mut new_updates = vec![];
+
+        if let Some(self_idx) = graph.get_node_index_by_id_opt(self.id()) {
+            if let Some(NodeWeight::InputSocket(InputSocketNodeWeight::V1(input_socket_inner))) =
+                graph
+                    .edges_directed(self_idx, Incoming)
+                    .filter(|edge_ref| {
+                        EdgeWeightKindDiscriminants::Prototype == edge_ref.weight().kind().into()
+                    })
+                    .filter_map(|edge_ref| graph.get_node_weight_opt(edge_ref.source()))
+                    .next()
```

Comment:
It's definitely adding on to the method call chaining going on, but if you use the "get node weight of kind" & `SiVersionedNodeWeight` trait here, you won't have to hard-code what version the inner is.

Something like:
```suggestion
            if let Ok(input_socket_inner) =
                graph
                    .edges_directed(self_idx, Incoming)
                    .filter(|edge_ref| {
                        EdgeWeightKindDiscriminants::Prototype == edge_ref.weight().kind().into()
                    })
                    .filter_map(|edge_ref| graph.get_node_weight_opt(edge_ref.source()))
                    .next()
                    .get_input_socket_node_weight()
                    .map(|nw| node_weight::traits::SiVersionedNodeWeight::inner(nw))
```

Reply:
ah yes, i forgot about our new traits